python/pyspark/pandas/sql_formatter.py

```diff
@@ -200,7 +201,8 @@ def sql(
     try:
         sdf = session.sql(formatter.format(query, **kwargs), args)
     finally:
-        formatter.clear()
+        if not is_remote():
+            formatter.clear()
```

Comment:
cc @zhengruifeng

Reply:
We should still clean them no?

Reply:
IMHO it is okay for now since the possibility of side effects is significantly low enough to ignore, since the name of temp view is generated by using `uuid.uuid4()` so it's almost no chance of overlap, and also all the temp views are automatically deleted when the session ends.

Yeah, but I still agree that ideally it would be best to ensure that the temp view operates properly even after it is deleted from the remote session.

Reply:
Nah, it will pollute all the catalog. Think about the case like `SHOW TABLES`, and it shows a bunch of random tables.

Reply:
> it shows a bunch of random tables.

Ah, I see. Then let's wait for another look from @zhengruifeng 👀 

And maybe we could just prevent using pandas-on-Spark object for `ps.sql` with Spark Connect for now, and document why we currently doesn't support this.

Reply:
It is a known issue, we cann't drop the temp view based on existing framework, and it will pollute the catalog.

Reply:
Thanks for checking!

Then I think maybe we can choose one of:
1. Support with proper warning that the catalog may be polluted due to a random temp view.
2. Not support with proper note why we currently don't support.

WDYT?